site/src/pages/[shortname]/[blockSlug].page.tsx

```diff
@@ -63,19 +67,13 @@ type BlockPageQueryParams = {
   blockSlug?: string[];
 };
 
-export const getStaticPaths: GetStaticPaths<
-  BlockPageQueryParams
-> = async () => {
+export const getStaticPaths: GetStaticPaths<BlockPageQueryParams> = () => {
   return {
-    paths: [],
+    paths: readBlocksFromDisk().map((metadata) => `/${metadata.packagePath}`),
```

Comment:
**Blocking:** Right now we can afford to read all the blocks from disk at build time, but the idea behind `fallback: "blocking"` is that we _don't_ determine the paths at build time. Therefore I suggest leaving this as:

```suggestion
    paths: [],
```

Reply:
We can determine paths to be built while using `fallback: blocking`, only difference is that instead of immediately returning a 404 for paths that don't exist there, it tries to generate the page.

The reason we set the paths is that we can't access the `blocks` directory at runtime. As a result, the explore blocks section is always empty because readBlocksFromApi (which tries to access the blocks directory) returns an empty array. The block page was rendering correct information previously because we were making [requests](https://github.com/blockprotocol/blockprotocol/blob/main/site/src/pages/%5Bshortname%5D/%5BblockSlug%5D.page.tsx#L138-L145) to fetch the respective info from the public folder instead of accessing them from the filesystem. Setting the paths ensures each block page specified is built at build time.


I understand building all the block pages at build time is not a great idea when we have a huge list of blocks. But I think the ideal situation will be to build common block pages (specified in the path list) at build time, then build and cache the remaining blocks as the user requests for them. It will be much easier to access the file system at runtime once we upgrade to [Next 12]() though.

For now, I think we can work with building all the block pages at build time. In an example Nextjs gives about Incremental Static Regeneration, you can build up to 1k product pages and generate the rest on demand. I doubt we'll notice any serious difference in build times for the time being. If we do, we can always try to improve it then

Reply:
I don't think it affects your discussion about building _pages_, but worth noting that the 'metadata and block package files in GitHub repo' approach is a temporary implementation. 

We'll store metadata in a database and the package files in some external object store.

Reply:
Thanks both, after more thought I agree @teenoh we can leave this as is for now. Perhaps a `@todo` about generating the paths for a limited subset of blocks would be good, but agree we should address this when we have a more long-term implementation of how we're storing the medatadata.

Reply:
Ahh I see, I didn't consider that @CiaranMn.  Thanks all, will add a `@todo`